src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.ts

```diff
@@ -48,71 +57,64 @@ export const AnalysisDuplicator = ({ destroyOld = false, fromLauncher = false, p
     okButton: h(ButtonPrimary, {
       disabled: errors || processing,
       tooltip: Utils.summarizeErrors(errors),
-      onClick: async () => {
+      // @ts-expect-error
+      onClick: withErrorReportingInModal(`Error ${destroyOld ? 'renaming' : 'copying'} analysis`, onDismiss, async () => {
         setProcessing(true)
-        try {
-          const rename = !!googleProject ?
-            () => Ajax().Buckets.analysis(googleProject, bucketName, printName, toolLabel).rename(newName) :
-            () => Ajax().AzureStorage.blob(workspaceId, printName).rename(newName)
+        const rename = isGoogleWorkspaceInfo(workspaceInfo) ?
+          () => Ajax().Buckets.analysis(workspaceInfo.googleProject, workspaceInfo.bucketName, printName, toolLabel).rename(newName) :
+          () => Ajax().AzureStorage.blob(workspaceInfo.workspaceId, printName).rename(newName)
 
-          const duplicate = !!googleProject ?
-            () => Ajax().Buckets.analysis(googleProject, bucketName, getFileName(printName), toolLabel).copy(`${newName}.${getExtension(printName)}`, bucketName, true) :
-            () => Ajax().AzureStorage.blob(workspaceId, printName).copy(newName)
+        const duplicate = isGoogleWorkspaceInfo(workspaceInfo) ?
+          () => Ajax().Buckets.analysis(workspaceInfo.googleProject, workspaceInfo.bucketName, getFileName(printName), toolLabel).copy(`${newName}.${getExtension(printName)}`, workspaceInfo.bucketName, true) :
+          () => Ajax().AzureStorage.blob(workspaceInfo.workspaceId, printName).copy(newName)
 
-          if (destroyOld) {
-            await rename()
-          } else {
-            await duplicate()
-          }
+        if (destroyOld) {
+          await rename()
+          // @ts-expect-error
+          await Ajax().Metrics.captureEvent(Events.notebookRename, {
```

Comment:
`captureEvent` doesn't need to be awaited.

Reply:
it doesnt semantically, by the linter complains if its not. I'll remove it for now...

Reply:
There are a lot of examples of unawaited captureEvent calls throughout Terra UI. I know TS complains because it doesn't think captureEvent is callable, but I don't think ESLint has any issue with this.